### PR TITLE
[weight] Support weight decay

### PR DESCRIPTION
**Pull request by kparichay**
- Add support for weight decay property which will enable decay of weights
with each applying of the gradient.
Weight decay can be enabled individually for both weight and bias.
This is kept separate from regularizer as they both behave differently.
- Add support for weight decay to the layers with weights.
Further update the requestWeight API of the layer context to accept
weight decay, and update the usage at manager.
- added unittest for weight decay along with fix.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM, except unittest for weight decay will be added I suppose.

**Review by lhs8928 — APPROVED** [review]
LGTM

**Comment by kparichay** [comment]
Added unittest for the weight decay.

File: nntrainer/tensor/weight.h [hunk 1]
```diff
@@ -262,11 +278,19 @@ class Weight : public Var_Grad {
...
+    1e-8; /**< epsilon for zero comparison */
 
   WeightRegularizer regularizer; /**< regularizer for this variable */
   float regularizer_constant;    /**< constant factor for regularization */
+  float decay;                   /**< constant factor for the weight decay */
   float clip_by_global_norm; /**< constant factor to clip gradient by L2 norm */
   std::vector<Tensor *> opt_vars; /**< optimizer variables */
+
+  /**
+   * @brief     Apply the gradient to the weight
```
[the remark this answers is not in the document]
Comment (kparichay) [reply nntrainer/tensor/weight.h]: done.

**Pull request merged**